src/token/stream-parser.ts

```diff
@@ -113,8 +175,17 @@ class Parser {
         parser.position += 1;
 
         if (type === TYPE.COLMETADATA) {
-          const token = await colMetadataParser(parser);
+          // console.log('>> entering colmetadataParser')
+          const token = await colMetadataParser(parser)
+          // console.log('>> done colMetadataParser, starting Promise.all(decryptSymmetricKey)..')
           parser.colMetadata = token.columns;
+          // console.log('colMetdata list = ', parser.colMetadata)
+          await Promise.all(parser.colMetadata.map(async (metadata) => {
+            if (metadata.cryptoMetadata) {
+              await decryptSymmetricKey(metadata.cryptoMetadata, options);
+            }
+          }));
```

Comment:
@arthurschreiber I also added these lines here, as this is where the decryption of the keys need to occur. I'm not sure if the other places for this decrypting process is really necessary though 🤔